src/main/java/com/keunsori/keunsoriserver/domain/reservation/service/ReservationService.java

```diff
@@ -82,4 +94,33 @@ public List<ReservationResponse> findAllMyReservations() {
         return reservationRepository.findAllByMember(member)
                 .stream().map(ReservationResponse::from).toList();
     }
+
+    public MonthlyScheduleResponse findMonthlySchedule(String yearMonth) {
+        List<DailyAvailableResponse> dailyAvailableResponses = findDailyAvailableByMonth(yearMonth);
+        List<ReservationResponse> reservationResponses = findReservationsByMonth(yearMonth);
+        return new MonthlyScheduleResponse(dailyAvailableResponses, reservationResponses);
+    }
+
+    private List<DailyAvailableResponse> findDailyAvailableByMonth(String yearMonth) {
+        List<DailyAvailableResponse> responses = new ArrayList<>();
+
+        LocalDate start = DateUtil.parseMonthToFirstDate(yearMonth);
+        LocalDate end = start.plusMonths(1);
+
+        Stream.iterate(start, date -> date.isBefore(end), date -> date.plusDays(1))
+                .forEach(
+                    date -> dailyScheduleRepository.findByDate(date)
+                            .ifPresentOrElse(
+                                    dailySchedule -> responses.add(DailyAvailableResponse.from(dailySchedule)),
+                                    () -> {
+                                        weeklyScheduleRepository.findByDayOfWeek(date.getDayOfWeek()).ifPresentOrElse(
+                                                weeklySchedule -> responses.add(DailyAvailableResponse.of(date, weeklySchedule)),
+                                                () -> responses.add(DailyAvailableResponse.createInactiveDate(date))
+                                        );
+                                    }
+                            )
+                );
```

Comment:
```suggestion
    private List<DailyAvailableResponse> findDailyAvailableByMonth(String yearMonth) {
        LocalDate start = DateUtil.parseMonthToFirstDate(yearMonth);
        LocalDate end = start.plusMonths(1);

        return Stream.iterate(start, date -> date.isBefore(end), date -> date.plusDays(1))
                .map(this::convertDateToDailyAvailableResponse).toList();
    }
    
    private DailyAvailableResponse convertDateToDailyAvailableResponse(LocalDate date) {
        return dailyScheduleRepository.findById(date)
                .map(DailyAvailableResponse::from)
                .orElseGet(() -> weeklyScheduleRepository.findByDayOfWeek(date.getDayOfWeek())
                        .map(schedule -> DailyAvailableResponse.of(date, schedule))
                        .orElseGet(() -> DailyAvailableResponse.createInactiveDate(date)));
    }
```

들여쓰기가 많아지는 것 같아서 혹시 요런 느낌은 어떤가요??